nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/pom.xml

```diff
@@ -468,6 +468,11 @@
             <groupId>org.glassfish.jaxb</groupId>
             <artifactId>jaxb-runtime</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-data-provenance-utils</artifactId>
+            <version>2.0.0-SNAPSHOT</version>
+        </dependency>
```

Comment:
Processors should not need to have a dependency on nifi-data-provenance-utils. The API should include anything necessary in order to emit the provenance event.

Reply:
We should probably eliminate this newly introduced `UploadContext`. While it does keep the `ProvenanceReporter` cleaner by avoiding introducing several potential overrides, it introduces more complexity into the Processors that might want to call the method. I.e., the overrides make the ProvenanceReporter more complex but make the Processors simpler. This is one of those situations where I could go either way. But the fact that interface already is laid out to use several overrides means it's cleaner to keep consistent. Additionally, it eliminates the need for something like this in data-provenance-utils and also avoids introducing something like this into the nifi-api